plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/CountingAccessFileHiveMetastore.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.plugin.hive.HdfsEnvironment;
+import io.trino.plugin.hive.NodeVersion;
+import io.trino.plugin.hive.PartitionStatistics;
+import io.trino.plugin.hive.authentication.HiveIdentity;
+import io.trino.plugin.hive.metastore.Database;
+import io.trino.plugin.hive.metastore.MetastoreConfig;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastore;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastoreConfig;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CountingAccessFileHiveMetastore
+        extends FileHiveMetastore
+{
+    public static final String GET_TABLE = "getTable";
+    public static final String GET_ALL_DATABASES = "getAllDatabases";
+    public static final String GET_DATABASE = "getDatabase";
+    public static final String GET_TABLE_WITH_PARAMETER = "getTablesWithParameter";
+    public static final String GET_TABLE_STATISTICS = "getTableStatistics";
```

Comment:
Let's define an enum and use it as a key in methodInvocations map

Reply:
ok